public/app/features/provisioning/ImportFromRepository.tsx

```diff
@@ -0,0 +1,72 @@
+import { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom-v5-compat';
+import { useAsync } from 'react-use';
+
+import { AppEvents } from '@grafana/data';
+import { config, getAppEvents, getBackendSrv } from '@grafana/runtime';
+import { Button, ConfirmModal } from '@grafana/ui';
+
+import { Loader } from '../plugins/admin/components/Loader';
+
+import { useCreateRepositoryImportMutation, useListRepositoryQuery } from './api';
+
+interface Props {
+  name?: string;
+  folder?: string;
+}
+
+export function ImportFromRepository({ name, folder }: Props) {
+  const query = useListRepositoryQuery();
+  const [importResource, importQuery] = useCreateRepositoryImportMutation();
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const navigate = useNavigate();
+
+  // TODO generate endpoints for this
+  const { value } = useAsync(async () => {
+    return getBackendSrv().get(`/apis/folder.grafana.app/v0alpha1/namespaces/${config.namespace}/folders/${folder}`);
+  });
+
+  useEffect(() => {
+    const appEvents = getAppEvents();
+    if (importQuery.isSuccess) {
+      appEvents.publish({
+        type: AppEvents.alertSuccess.name,
+        payload: ['Resources imported'],
+      });
+      // TODO the URL will be different based on the type of resource imported
+      navigate(`/dashboards${folder ? `/f/${folder}` : ''}`);
+    } else if (importQuery.isError) {
+      appEvents.publish({
+        type: AppEvents.alertError.name,
+        payload: ['Error importing resources', importQuery.error],
+      });
+    }
+  }, [folder, importQuery.error, importQuery.isError, importQuery.isSuccess, navigate]);
+
+  const onClick = () => {
+    if (!name) {
+      return;
+    }
+
+    importResource({ name, ref: 'main' });
```

Comment:
This cannot be hard-coded to the main branch because the repositories are configured with a branch (that part is also missing from)

![image](https://github.com/user-attachments/assets/aa0f4f52-e74c-4b78-90c7-8e0f94a0aedc)

Let's fix both in follow-up PRs because this button also helps everybody to test things a bit more conveniently.